lib/cdo/shared_constants.rb

```diff
@@ -109,17 +108,20 @@ module SharedConstants
     gamelab
   ).freeze
 
+  # For privacy reasons, sprite lab projects can only be published if they are not is "restricted share mode".
+  # This mode is triggered if the user is a student and has uploaded an image to their project.
+  RESTRICTED_PUBLISH_PROJECT_TYPES = %w(spritelab).freeze
+
   UNPUBLISHABLE_PROJECT_TYPES = %w(
     algebra_game
     calc
     eval
     minecraft_codebuilder
-    spritelab
```

Comment:
Out of curiosity, do you know why spritelab was part of both ALWAYS_PUBLISHABLE_PROJECT_TYPES and UNPUBLISHABLE_PROJECT_TYPES?

Reply:
I have no idea. I found the [PR](https://github.com/code-dot-org/code-dot-org/pull/25461/files) when we added it to ALWAYS_PUBLISHABLE and there was no note about it already being unpublishable, so my guess is it was an oversight.

Reply:
[this](https://github.com/code-dot-org/code-dot-org/pull/24766/files) is where unpublishable, it seems to only be used as part of ALL_PROJECT_TYPES